### PR TITLE
RMAC-8115 - Handle exception not observed

### PR DESCRIPTION
**Pull request by heldergoncalves92**
JIRA: https://outsystemsrd.atlassian.net/browse/RMAC-8115

### CODE REVIEW

File: CefGlue.Avalonia/CefGlue.Avalonia.csproj [hunk 1]
```diff
@@ -20,7 +20,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference PrivateAssets="all" Include="Avalonia" Version="0.10.6" />
+    <PackageReference PrivateAssets="all" Include="Avalonia" Version="0.10.7" />
```
Comment (heldergoncalves92) [inline CefGlue.Avalonia/CefGlue.Avalonia.csproj:23]: Not needed on fix (just to have a more recent version)

File: CefGlue.Demo.Avalonia/CefGlue.Demo.Avalonia.csproj [hunk 2]
```diff
@@ -44,9 +44,9 @@
   <Import Project="$(ProjectDir)..\CefGlue.CopyLocal.props" />
 
   <ItemGroup>
-    <PackageReference Include="Avalonia.Desktop" Version="0.10.6" />
+    <PackageReference Include="Avalonia.Desktop" Version="0.10.7" />
```
Comment (heldergoncalves92) [inline CefGlue.Demo.Avalonia/CefGlue.Demo.Avalonia.csproj:47]: Same. Not needed

File: CefGlue.Common/JavascriptExecution/JavascriptExecutionEngine.cs [hunk 3]
```diff
@@ -147,9 +147,13 @@ private T ProcessResult<T>(Task<object> task, int taskId, bool timedOut = false)
 
                 return JavascriptToNativeTypeConverter.ConvertToNative<T>(task.Result);
             }
-            catch
+            catch (Exception e)
             {
                 _pendingTasks.TryRemove(taskId, out var _);
+                
+                if (e is AggregateException && e.InnerException is TaskCanceledException) {
```
Comment (joaompneves) [inline CefGlue.Common/JavascriptExecution/JavascriptExecutionEngine.cs:154]: you should try to create a test for this

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs [hunk 4]
```diff
@@ -13,5 +14,31 @@ public async Task LoadCustomScheme()
             Browser.Address = $"{CustomSchemeHandlerFactory.SchemeName}://testdomain/test";
             await loadTask;
         }
+
+        [Test]
+        public async Task JavascriptExecutionEngine_DisposeBrowser_ShouldHandleInnerTaskCanceledExceptions()
```
Comment (joaompneves) [inline CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:19]: not a custom scheme test.. its a js test

Comment (joaompneves) [reply CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:19]: please follow the tests naming convention used

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs [hunk 5]
```diff
@@ -13,5 +14,31 @@ public async Task LoadCustomScheme()
...
+        {
+            // Arrange
+            Exception exception = null;
+            var t = Browser.EvaluateJavaScript<string>("function ff(){ let finishDate = new Date((new Date()).getTime() + 10000); let sum = 0; while(new Date() < finishDate) {sum++;} return sum;} ff()");
```
Comment (joaompneves) [inline CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:23]:
```suggestion
            var t = Browser.EvaluateJavaScript<string>("(() => { const finishTime = new Date().getTime() + 10000; while(new Date().getTime() < finishTime); return 10;})()");
```

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs [hunk 6]
```diff
@@ -13,5 +14,31 @@ public async Task LoadCustomScheme()
...
+                    await t;
+                }
+                catch (Exception e)
+                {
+                    exception = e;
+                }
+            });
+
+            // Act
+            var t2 = Task.Run(() => Browser.Dispose());
```
Comment (joaompneves) [inline CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:37]: why do you need a task to run dispose?

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs [hunk 7]
```diff
@@ -13,5 +14,31 @@ public async Task LoadCustomScheme()
...
+            Task.WaitAll(t1, t2);
+
+            // Assert
+            Assert.IsNull(exception);
```
Comment (joaompneves) [inline CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:41]: check that the value returned is the default

**Review by joaompneves — CHANGES_REQUESTED** [review]

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs (hunk 6 above)
Comment (heldergoncalves92) [reply CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:37]: Not needed after all

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs (hunk 5 above)
Comment (heldergoncalves92) [reply CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:23]: Done

File: CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs [hunk 8]
```diff
@@ -13,5 +14,31 @@ public async Task LoadCustomScheme()
...
+        }
```
[the remark this answers is not in the document]
Comment (heldergoncalves92) [reply CefGlue.Tests/CustomSchemes/CustomSchemesTests.cs:42]: We already have that test. Check: `CancelledOnTimeout`

File: CefGlue.Tests/Javascript/JavascriptEvaluationTests.cs [hunk 9]
```diff
@@ -79,5 +79,33 @@ public async Task NotCancelledBeforeTimeout()
...
+                }
+            });
+
+            // Act
+            Browser.Dispose();
+            t1.Wait();
+
+            // Assert
+            Assert.IsNull(exception);
+            Assert.AreEqual(default, result);
```
Comment (joaompneves) [inline CefGlue.Tests/Javascript/JavascriptEvaluationTests.cs:108]:
```suggestion
            // Arrange
            var evalTask = EvaluateJavascript<int>("const finishTime = new Date().getTime() + 10000; while(new Date().getTime() < finishTime); return 10;");

            // Act
            Browser.Dispose();
            
            try 
            {
                var result = await evalTask;
                Assert.AreEqual(0, result);
            }
            catch (Exception e)
            {
                Assert.Fail(e);
            }
```

**Review by joaompneves — APPROVED** [review]

**Review by os-lmo — APPROVED** [review]

**Pull request merged**